runtime: Kata runtime fails to build with go 1.8.3

# Description of problem
Build and install of Kata runtime fails

# Expected result
Successfully built the Kata runtime

# Actual result
```
runtime]# make && sudo -E PATH=$PATH make install
     CLEAN   clean
     GENERATE cli/config-generated.go
     CONFIG  data/kata-collect-data.sh
kata-runtime - version 1.2.0 (commit ed6f715ae6d46171c170c98909656539005766e3)

• architecture:
	Host: x86_64
	golang: amd64
	Build: amd64

• golang:
	go version go1.8.3 linux/amd64

• Summary:

	destination install path (DESTDIR)    : /
	binary installation path (BINDIR)     : /usr/local/bin
	binaries to install                   :
	 - /usr/local/bin/kata-runtime
	 - /usr/local/bin/data/kata-collect-data.sh
	config to install (CONFIG)            : cli/config/configuration.toml
	install path (CONFIG_PATH)            : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (SYSCONFIG)     : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-lite-system-x86_64
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers

     BUILD   /home/nitesh/TST/kata/src/github.com/kata-containers/runtime/kata-runtime
# github.com/kata-containers/runtime/virtcontainers
../virtcontainers/agent.go:94: cannot use kataAgent literal (type *kataAgent) as type agent in return argument:
	*kataAgent does not implement agent (wrong type for init method)
		have init("github.com/kata-containers/runtime/vendor/golang.org/x/net/context".Context, *Sandbox, interface {}) error
		want init("context".Context, *Sandbox, interface {}) error
make: *** [Makefile:331: /home/nitesh/TST/kata/src/github.com/kata-containers/runtime/kata-runtime] Error 
```

No, I do not have any local changes.
```
[root@nitkon runtime]# dir="/tmp/go"
[root@nitkon runtime]# url="github.com/kata-containers/runtime"
[root@nitkon runtime]# GOPATH="$dir" go get -d "$url"
package github.com/kata-containers/runtime: no buildable Go source files in /tmp/go/src/github.com/kata-containers/runtime
[root@nitkon runtime]# cd "$dir/src/$url"
[root@nitkon runtime]# GOPATH="$dir" make
     CLEAN   clean
     GENERATE cli/config-generated.go
     CONFIG  data/kata-collect-data.sh
kata-runtime - version 1.2.0 (commit 8b5fa32d9b55b8a456807fe3e7fe527633bf389b)

• architecture:
	Host: x86_64
	golang: amd64
	Build: amd64

• golang:
	go version go1.8.3 linux/amd64

• Summary:

	destination install path (DESTDIR)    : /
	binary installation path (BINDIR)     : /usr/local/bin
	binaries to install                   :
	 - /usr/local/bin/kata-runtime
	 - /usr/local/bin/data/kata-collect-data.sh
	config to install (CONFIG)            : cli/config/configuration.toml
	install path (CONFIG_PATH)            : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (SYSCONFIG)     : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-lite-system-x86_64
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers

     BUILD   /tmp/go/src/github.com/kata-containers/runtime/kata-runtime
# github.com/kata-containers/runtime/virtcontainers
../virtcontainers/agent.go:94: cannot use kataAgent literal (type *kataAgent) as type agent in return argument:
	*kataAgent does not implement agent (wrong type for init method)
		have init("github.com/kata-containers/runtime/vendor/golang.org/x/net/context".Context, *Sandbox, interface {}) error
		want init("context".Context, *Sandbox, interface {}) error
make: *** [Makefile:331: /tmp/go/src/github.com/kata-containers/runtime/kata-runtime] Error 2

```

@devimc : So kata runtime with golang version 1.9.7 builds fine? I might have a local setup issue then ...

```
[root@nitkon runtime]# export GOROOT=/usr/local/go
[root@nitkon runtime]# export GOPATH=/home/nitesh/TST/kata/
[root@nitkon runtime]# export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
[root@nitkon runtime]# go version
go version go1.9.7 linux/amd64
[root@nitkon runtime]# cat versions.yaml | grep "Google's 'go' language" -A2 -B2

  golang:
    description: "Google's 'go' language"
    notes: "'version' is the default minimum version used by this project."
    version: "1.9.7"
[root@nitkon runtime]# PWD
bash: PWD: command not found...
Similar command is: 'pwd'
[root@nitkon runtime]# pwd
/home/nitesh/TST/kata/src/github.com/kata-containers/runtime
[root@nitkon runtime]# make && sudo -E PATH=$PATH make install
     CLEAN   clean
     GENERATE cli/config-generated.go
     CONFIG  data/kata-collect-data.sh
kata-runtime - version 1.2.0 (commit ed6f715ae6d46171c170c98909656539005766e3-dirty)

• architecture:
	Host: x86_64
	golang: amd64
	Build: amd64

• golang:
	go version go1.9.7 linux/amd64

• Summary:

	destination install path (DESTDIR)    : /
	binary installation path (BINDIR)     : /usr/local/bin
	binaries to install                   :
	 - /usr/local/bin/kata-runtime
	 - /usr/local/bin/data/kata-collect-data.sh
	config to install (CONFIG)            : cli/config/configuration.toml
	install path (CONFIG_PATH)            : /usr/share/defaults/kata-containers/configuration.toml
	alternate config path (SYSCONFIG)     : /etc/kata-containers/configuration.toml
	hypervisor path (QEMUPATH)            : /usr/bin/qemu-lite-system-x86_64
	assets path (PKGDATADIR)              : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR)       : /usr/libexec/kata-containers

     BUILD   /home/nitesh/TST/kata/src/github.com/kata-containers/runtime/kata-runtime
# runtime
/usr/local/go/src/runtime/mstkbar.go:151:10: debug.gcstackbarrieroff undefined (type struct { allocfreetrace int32; cgocheck int32; efence int32; gccheckmark int32; gcpacertrace int32; gcshrinkstackoff int32; gcrescanstacks int32; gcstoptheworld int32; gctrace int32; invalidptr int32; sbrk int32; scavenge int32; scheddetail int32; schedtrace int32 } has no field or method gcstackbarrieroff)
/usr/local/go/src/runtime/mstkbar.go:162:24: division by zero
/usr/local/go/src/runtime/mstkbar.go:162:43: invalid expression unsafe.Sizeof(composite literal)
/usr/local/go/src/runtime/mstkbar.go:162:44: undefined: stkbar
/usr/local/go/src/runtime/mstkbar.go:212:4: gp.stkbar undefined (type *g has no field or method stkbar)
/usr/local/go/src/runtime/mstkbar.go:213:15: gp.stkbar undefined (type *g has no field or method stkbar)
/usr/local/go/src/runtime/mstkbar.go:216:23: undefined: stackBarrierPC
/usr/local/go/src/runtime/mstkbar.go:226:28: gp.stkbarPos undefined (type *g has no field or method stkbarPos)
/usr/local/go/src/runtime/mstkbar.go:227:19: gp.stkbarPos undefined (type *g has no field or method stkbarPos)
/usr/local/go/src/runtime/mstkbar.go:248:41: undefined: stkbar
/usr/local/go/src/runtime/mstkbar.go:227:19: too many errors
make: *** [Makefile:331: /home/nitesh/TST/kata/src/github.com/kata-containers/runtime/kata-runtime] Error 2
```

--- a/virtcontainers/agent.go
+++ b/virtcontainers/agent.go
@@ -6,13 +6,13 @@
 package virtcontainers
 
 import (
-	"context"
 	"fmt"
 	"syscall"
 
 	"github.com/kata-containers/agent/protocols/grpc"
 	"github.com/mitchellh/mapstructure"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"golang.org/x/net/context"
 )
 
 // AgentType describes the type of guest agent a Sandbox should run.

--- a/virtcontainers/hyperstart_agent.go
+++ b/virtcontainers/hyperstart_agent.go
@@ -6,7 +6,6 @@
 package virtcontainers
 
 import (
-	"context"
 	"fmt"
 	"net"
 	"net/url"
@@ -25,6 +24,7 @@ import (
 	ns "github.com/kata-containers/runtime/virtcontainers/pkg/nsenter"
 	"github.com/kata-containers/runtime/virtcontainers/utils"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"golang.org/x/net/context"
 )
 
 var defaultSockPathTemplates = []string{"%s/%s/hyper.sock", "%s/%s/tty.sock"}

--- a/virtcontainers/noop_agent.go
+++ b/virtcontainers/noop_agent.go
@@ -6,11 +6,11 @@
 package virtcontainers
 
 import (
-	"context"
 	"syscall"
 
 	"github.com/kata-containers/agent/protocols/grpc"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"golang.org/x/net/context"
 )
 
 // noopAgent a.k.a. NO-OP Agent is an empty Agent implementation, for testing and

